src/main/java/org/mapstruct/intellij/codeinsight/references/MapstructMappingQualifiedByNameReference.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.intellij.codeinsight.references;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.intellij.codeInsight.lookup.LookupElement;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.PsiType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mapstruct.intellij.util.MapstructAnnotationUtils;
+import org.mapstruct.intellij.util.MapstructUtil;
+
+import static com.intellij.codeInsight.AnnotationUtil.findAnnotation;
+import static com.intellij.codeInsight.AnnotationUtil.getStringAttributeValue;
+import static org.mapstruct.intellij.util.MapstructUtil.asLookupWithRepresentableText;
+
+/**
+ * Reference for {@link org.mapstruct.Mapping#qualifiedByName()}.
+ *
+ * @author Oliver Erhart
+ */
+class MapstructMappingQualifiedByNameReference extends MapstructBaseReference {
+
+    /**
+     * Create a new {@link MapstructMappingQualifiedByNameReference} with the provided parameters
+     *
+     * @param element the element that the reference belongs to
+     * @param previousReference the previous reference if there is one (in nested properties for example)
+     * @param rangeInElement the range that the reference represent in the {@code element}
+     * @param value the matched value (useful when {@code rangeInElement} is empty)
+     */
+    private MapstructMappingQualifiedByNameReference(PsiElement element,
+                                                     MapstructMappingQualifiedByNameReference previousReference,
+                                                     TextRange rangeInElement, String value) {
+        super( element, previousReference, rangeInElement, value );
+    }
+
+    @Override
+    PsiElement resolveInternal(@NotNull String value, @NotNull PsiType psiType) {
+        return null; // not needed
+    }
+
+    @Override
+    PsiElement resolveInternal(@NotNull String value, @NotNull PsiMethod mappingMethod) {
+
+        List<PsiMethod> allMethodsFromThisAndReferencedMappers = findAllMethodsFromThisAndReferencedMappers(
+            mappingMethod );
+
+        return allMethodsFromThisAndReferencedMappers.stream()
+            .filter( MapstructUtil::isNamedMethod )
+            .filter( this::methodHasReturnType )
```

Comment:
Let's move these checks in the `findAllMethodsFromThisAndReferencedMappers`.  That one is already partially doing this for the named methods in used mappers.